File: performance_tests/string_pos.cpp
#include <iostream>
#include <string>

size_t foo(size_t x, const std::string& s) {
    // const std::string s = "testing";

    const auto a = s.find('s');
    const auto b = a != std::string::npos ? s.find('e') : std::string::npos;
    const auto c = b != std::string::npos ? s.find('i') : std::string::npos;

    return b != std::string::npos ? x + a + b + c : b != std::string::npos;
}

int main() {
    std::string s;

    std::cin >> s;

    size_t x = 0;
    for (size_t i = 0; i < 100'000'000; ++i) {
        const size_t y = foo(x, s);
        if (y != std::string::npos) {
            x = y;
        }
    }

    std::cout << x << std::endl;

    return 0;
}

File: performance_tests/string_pos.rs
use std::io;

pub fn foo(x: usize, s: &str) -> Option<usize> {
    // let s = "testing";
    // let ss = s.to_string();
    Some(x + s.find('s')? + s.find('e')? + s.find('i')?)
}

fn main() {
    let mut line = String::new();
    io::stdin().read_line(&mut line).unwrap();

    let mut x: usize = 0;
    for _i in 0..100_000_000 {
        x = foo(x, &line).unwrap_or(x);
    }
    
    println!("{}", x)
}
